runtime/autoload/lsp/server.vim

```diff
@@ -0,0 +1,91 @@
+let s:server_configuration = get(s:, 'server_configuration', {})
```

Comment:
Wouldn't it be simpler if this entire module was in lua? Having lots of small accessor functions both here and in lua seems weird, config should be available as a dict/table where it is used.